src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -510,3 +510,68 @@ export const handleDuplicateAdminForm: RequestHandler<
       })
   )
 }
+
+/**
+ * Handler for POST /:formId/adminform/copy
+ * Duplicates the form corresponding to the formId. The form must be public to
+ * be copied.
+ * @note The current user will be the admin of the new duplicated form
+ * @security session
+ *
+ * @returns 200 with the duplicate form dashboard view
+ * @returns 403 when form is private
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handleCopyTemplateForm: RequestHandler<
+  { formId: string },
+  unknown,
+  DuplicateFormBody
+> = (req, res) => {
+  const { formId } = req.params
+  const userId = (req.session as Express.AuthedSession).user._id
+  const overrideParams = req.body
+
+  // TODO(#792): Remove when frontend has stopped sending isTemplate.
+  if ('isTemplate' in req.body) {
+    logger.info({
+      message: 'isTemplate is still being sent by the frontend',
+      meta: {
+        action: 'handleCopyTemplateForm',
+      },
+    })
+  }
+
+  return (
+    // Step 1: Retrieve currently logged in user.
+    UserService.getPopulatedUserById(userId)
+      .andThen((user) =>
+        // Step 2: Check if form is currently public.
+        AuthService.getFormIfPublic(formId).andThen((originalForm) =>
+          // Step 3: Duplicate form.
+          duplicateForm(originalForm, userId, overrideParams)
```

Comment:
do you think it might be better to un-nest these as far as possible? a lot easier to understand chaining than nesting